test/ardop/test_ARDOPC.c

```diff
@@ -0,0 +1,131 @@
+{
+	UCHAR compressed[6];
+	char compressed_hex[2*sizeof(compressed) + 1];
+	compressed_hex[2 * sizeof(compressed)] = '\0';
+
+	// check string must be exactly 12 characters long
+	assert_int_equal(strlen(hexstr), 2 * sizeof(compressed));
+
+	CompressCallsign(call, compressed);
+
+	for (size_t i = 0; i < sizeof(compressed); ++i) {
+		snprintf(&compressed_hex[2*i], 3, "%02x", (unsigned int)compressed[i]);
+	}
+	assert_string_equal(hexstr, compressed_hex);
+}
+
+/*
+ * Verify that the given `call_in` callsign (with optional SSID) becomes
+ * `call_out` after a round-trip through the callsign compress/decompress
+ * algorithm. This verifies that the callsign can be sent and received.
+ */
+static void assert_callsign_inout(const char *call_in, const char *call_out)
+{
+	char out[11];
+	UCHAR compressed[6];
+
+	CompressCallsign(call_in, compressed);
+	DeCompressCallsign(compressed, out, sizeof(out));
+	assert_string_equal(call_out, out);
+}
+
+/* test string-packing compression and decompression */
+static void test_ascii_6bit(void **state)
+{
+	(void)state; /* unused */
+
+	assert_6bit_inout("        ", "        ");
+	assert_6bit_inout("AZ09-_  ", "AZ09-_  ");
+	assert_6bit_inout("A      Z", "A      Z");
+	assert_6bit_inout("HI\0\0\0\0\0U", "HI      ");
+	assert_6bit_inout("abcxyz  ", "ABCXYZ  ");
+}
+
+/* test callsign wireline representation */
+static void test_compress_callsign(void **state)
+{
+	(void)state; /* unused */
+
+	// six character + SSID values are taken from direct
+	// invocation of CompressCallsign() in ardopcf eab1f3165a30.
+	assert_callsign_wireline_is("A1A", "851840000010");
+	assert_callsign_wireline_is("A1A-1", "851840000011");
+	assert_callsign_wireline_is("N0CALL", "b908e1b2c010");
+	assert_callsign_wireline_is("N0CALL-A", "b908e1b2c021");
+	assert_callsign_wireline_is("N0CALL-Z", "b908e1b2c03a");
+
+	// non-canonical representation
+	assert_callsign_wireline_is("N0CALL-0", "b908e1b2c010");
+}
+
+/* test callsign round-trip through the protocol */
+static void test_decompress_callsign(void **state)
+{
+	(void)state; /* unused */
+
+	assert_callsign_inout("A1A", "A1A");
+	assert_callsign_inout("A1A-1", "A1A-1");
+	assert_callsign_inout("N0CALL", "N0CALL");
+	assert_callsign_inout("N0CALL-A", "N0CALL-A");
+	assert_callsign_inout("N0CALL-Z", "N0CALL-Z");
+	assert_callsign_inout("LONGCAL-15", "LONGCAL-15");
+
+	// non-canonical representation
+	assert_callsign_inout("N0CALL-0", "N0CALL");
+
+	// too long callsign
+	assert_callsign_inout("CAFECAFE", "CAFECAF");
+
+	// overflow
+	assert_callsign_inout("MUCHTOOLON-G", "MUCHTOO");
+
+	// empty
+	assert_callsign_inout("", "");
+	assert_callsign_inout(NULL, "");
+}
+
+int main(void)
+{
+	const struct CMUnitTest tests[] = {
+		cmocka_unit_test(test_ascii_6bit),
+		cmocka_unit_test(test_compress_callsign),
+		cmocka_unit_test(test_decompress_callsign),
+	};
+
+	return cmocka_run_group_tests(tests, NULL, NULL);
```

Comment:
I had not considered the possibility of parallelization of tests.  Based on that, I agree that logging to file and console should be suppressed.  This can easily be achieved by adding
```
extern int FileLogLevel;
extern int ConsoleLogLevel;
```
to the top of each `test_*.c` file and adding 
```
        FileLogLevel = 0;
        ConsoleLogLevel = 0;
```
to the start of `main()` in each of those files.  This will suppress all logging except for the most extreme LOGEMERGENCY, which is not currently used anywhere in this code base.

As suggested by my earlier comment, this also leaves it easy to temporarily comment out these settings while trying to better understand a failed test condition.  I prefer this to the use of environment variables, which can potentially complicate cross-platform consistency.

Reply:
> and adding
> 
>         FileLogLevel = 0;
>         ConsoleLogLevel = 0;
> 
> to the start of main() in each of those files

Counterproposal: let's collect any common initialization in one place. 9d50e2a9ddc554e5a61f39f0aef59051563adb45 introduces `ardop_test_setup()` for all test-case executables.

cmocka has "group setup" and "group teardown" functions, but I'd rather reserve those for executable-specific behavior.

>  I prefer this to the use of environment variables, which can potentially complicate cross-platform consistency

Surprisingly, environment variables aren't all *that* different across platforms. Even Windows has them, and they work pretty much like their Linux counterparts.

Reply:
That works.